### ISSUE TITLE
statically linked version of nw for Linux

### ISSUE THREAD

**Issue by gaubert** [nwjs/nw.js#182]
First I really like node-webkit even if it is still a bit too young. Great work !!!.

It would be interesting to have a statically linked version of nw because depending on which flavour and version of linux dist you run you might get some issue. 
I get errors like:
./nw: /lib/libc.so.6: version `GLIBC_2.11' not found (required by ./nw)
./nw: /usr/lib/libstdc++.so.6: version`GLIBCXX_3.4.11' not found (required by ./nw)

nw should not require the user to install newer libc and libstdc++ for running.
It should work out of the box otherwise why providing an executable for nw.

It is not ideal but the easiest solution for that is to statically link everything (or at least part of it). 
It will make a big executable but at least it will work.

!$ ldd nw
./nw: /lib/libc.so.6: version `GLIBC_2.11' not found (required by ./nw)
./nw: /usr/lib/libstdc++.so.6: version`GLIBCXX_3.4.11' not found (required by ./nw)
        linux-gate.so.1 =>  (0xffffe000)
        libX11.so.6 => /usr/lib/libX11.so.6 (0xb75c9000)
        libXrandr.so.2 => /usr/lib/libXrandr.so.2 (0xb75c1000)
        librt.so.1 => /lib/librt.so.1 (0xb75b7000)
        libdl.so.2 => /lib/libdl.so.2 (0xb75b1000)
        libgobject-2.0.so.0 => /usr/lib/libgobject-2.0.so.0 (0xb756e000)
        libgthread-2.0.so.0 => /usr/lib/libgthread-2.0.so.0 (0xb7568000)
        libglib-2.0.so.0 => /usr/lib/libglib-2.0.so.0 (0xb74a3000)
        libgtk-x11-2.0.so.0 => /usr/lib/libgtk-x11-2.0.so.0 (0xb70ad000)
        libgdk-x11-2.0.so.0 => /usr/lib/libgdk-x11-2.0.so.0 (0xb7014000)
        libatk-1.0.so.0 => /usr/lib/libatk-1.0.so.0 (0xb6ff6000)
        libgdk_pixbuf-2.0.so.0 => /usr/lib/libgdk_pixbuf-2.0.so.0 (0xb6fda000)
        libpangocairo-1.0.so.0 => /usr/lib/libpangocairo-1.0.so.0 (0xb6fce000)
        libcairo.so.2 => /usr/lib/libcairo.so.2 (0xb6f4f000)
        libpango-1.0.so.0 => /usr/lib/libpango-1.0.so.0 (0xb6f07000)
        libfreetype.so.6 => /usr/lib/libfreetype.so.6 (0xb6e86000)
        libfontconfig.so.1 => /usr/lib/libfontconfig.so.1 (0xb6e55000)
        libXcomposite.so.1 => /usr/lib/libXcomposite.so.1 (0xb6e51000)
        libasound.so.2 => /usr/lib/libasound.so.2 (0xb6d7e000)
        libnss3.so => /usr/lib/libnss3.so (0xb6c50000)
        libnssutil3.so => /usr/lib/libnssutil3.so (0xb6c35000)
        libsmime3.so => /usr/lib/libsmime3.so (0xb6c0a000)
        libplc4.so => /usr/lib/libplc4.so (0xb6c04000)
        libnspr4.so => /usr/lib/libnspr4.so (0xb6bc7000)
        libpthread.so.0 => /lib/libpthread.so.0 (0xb6bad000)
        libXrender.so.1 => /usr/lib/libXrender.so.1 (0xb6ba3000)
        libgconf-2.so.4 => /usr/lib/libgconf-2.so.4 (0xb6b6b000)
        libdbus-1.so.3 => /lib/libdbus-1.so.3 (0xb6b28000)
        libXext.so.6 => /usr/lib/libXext.so.6 (0xb6b17000)
        libXfixes.so.3 => /usr/lib/libXfixes.so.3 (0xb6b11000)
        libudev.so.0 => /lib/libudev.so.0 (0xb6b07000)
        libexpat.so.1 => /lib/libexpat.so.1 (0xb6adf000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0xb69eb000)
        libm.so.6 => /lib/libm.so.6 (0xb69c1000)
        libgcc_s.so.1 => /lib/libgcc_s.so.1 (0xb69b2000)
        libc.so.6 => /lib/libc.so.6 (0xb6856000)
        libxcb-xlib.so.0 => /usr/lib/libxcb-xlib.so.0 (0xb6853000)
        libxcb.so.1 => /usr/lib/libxcb.so.1 (0xb6835000)
        libXau.so.6 => /usr/lib/libXau.so.6 (0xb6831000)
        /lib/ld-linux.so.2 (0xb7724000)
        libpcre.so.0 => /usr/lib/libpcre.so.0 (0xb67fd000)
        libXinerama.so.1 => /usr/lib/libXinerama.so.1 (0xb67f9000)
        libXi.so.6 => /usr/lib/libXi.so.6 (0xb67ef000)
        libXcursor.so.1 => /usr/lib/libXcursor.so.1 (0xb67e4000)
        libXdamage.so.1 => /usr/lib/libXdamage.so.1 (0xb67e0000)
        libpixman-1.so.0 => /usr/lib/libpixman-1.so.0 (0xb6797000)
        libpng12.so.0 => /usr/lib/libpng12.so.0 (0xb676d000)
        libxcb-render-util.so.0 => /usr/lib/libxcb-render-util.so.0 (0xb6768000)
        libxcb-render.so.0 => /usr/lib/libxcb-render.so.0 (0xb675f000)
        libgio-2.0.so.0 => /usr/lib/libgio-2.0.so.0 (0xb66ea000)
        libpangoft2-1.0.so.0 => /usr/lib/libpangoft2-1.0.so.0 (0xb66be000)
        libz.so.1 => /lib/libz.so.1 (0xb66a9000)
        libgmodule-2.0.so.0 => /usr/lib/libgmodule-2.0.so.0 (0xb66a4000)
        libplds4.so => /usr/lib/libplds4.so (0xb669f000)
        libORBit-2.so.0 => /usr/lib/libORBit-2.so.0 (0xb6642000)
        libdbus-glib-1.so.2 => /usr/lib/libdbus-glib-1.so.2 (0xb6622000)
        libnsl.so.1 => /lib/libnsl.so.1 (0xb6609000)
        libselinux.so.1 => /lib/libselinux.so.1 (0xb65ec000)

**Comment by Nosfy01** [comment]
Hello, we are in september 2018. I'm running Centos 7 and I get the following error:
Failed to load node library (error: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /usr/lib/node_modules/nw/nwjs/lib/libnode.so)).

I know absolutely nothing about all those dependencies (there name seem to start with 'g' probably for 'gnu' - lib for library and c for the language... I'm telling you this, hoping you understand that questioning me on what version of whatever lib or .lib.so or glib something I have installed will not help).

I tried [this](https://stackoverflow.com/questions/44205687/glibcxx-3-4-21-not-found-on-centos-7#). No success.

I know this though. 
My system is up to date.
As of now, the most recent version of node (10.9.0) runs perfectly well on my system.
The most recent version of chrome (68.0.3440.106) also.

It is hard for me to understand why nwjs can't run...

What specific linux distro should we use if we want nwjs?
